src/main/java/org/reactome/release/cosmicupdate/COSMICUpdateUtil.java

```diff
@@ -0,0 +1,320 @@
+	 * If you plan to add more "COSMIC" ReferenceDatabase objects, this code will need to be changed to use the <em>correct</em> "COSMIC"
+	 * ReferenceDatabase.
+	 * @param adaptor
+	 * @return A Collection of DatabaseIdentifier objects.
+	 * @throws SQLException
+	 * @throws Exception
+	 * @throws InvalidAttributeException
+	 */
+	static Collection<GKInstance> getCOSMICIdentifiers(MySQLAdaptor adaptor) throws SQLException, Exception, InvalidAttributeException
+	{
+		@SuppressWarnings("unchecked")
+		Collection<GKInstance> refDBs = adaptor.fetchInstanceByAttribute(ReactomeJavaConstants.ReferenceDatabase, ReactomeJavaConstants.name, " = ", "COSMIC");
+		GKInstance cosmicRefDB = null;
+		if (refDBs.size() == 1)
+		{
+			cosmicRefDB = refDBs.stream().findFirst().get();
+		}
+		else
+		{
+			logger.error("Wrong number of COSMIC refDBs: {} ; only 1 was expected.", refDBs.size());
+			logger.error("Cannot proceeed!");
+			System.exit(1);
+		}
+		
+		@SuppressWarnings("unchecked")
+		Collection<GKInstance> cosmicObjects = adaptor.fetchInstanceByAttribute(ReactomeJavaConstants.DatabaseIdentifier, ReactomeJavaConstants.referenceDatabase, " = ", cosmicRefDB.getAttributeValue(ReactomeJavaConstants.DB_ID));
+		return cosmicObjects;
+	}
+	
+	
+	/**
+	 * Produces a report on identifiers. Report indicates old/"legacy" identifiers, suggested prefixes, new identifiers suggested from COSMIC files,
+	 * and validity of old identifiers.
+	 * @param updaters The map of identifier updaters.
+	 * @throws IOException
+	 */
+	public static void printIdentifierUpdateReport(Map<String, COSMICIdentifierUpdater> updaters) throws IOException
+	{
+		// Create the reports directory if it's missing.
+		if (!Files.exists(Paths.get(COSMICUpdateUtil.reportsDirectoryPath)))
+		{
+			Files.createDirectories(Paths.get(COSMICUpdateUtil.reportsDirectoryPath));
+		}
+		try(CSVPrinter printer = new CSVPrinter(new FileWriter(COSMICUpdateUtil.reportsDirectoryPath + "/COSMIC-identifiers-report_"+dateSuffix+".csv"), CSVFormat.DEFAULT.withHeader("DB_ID", "Identifier", "Suggested Prefix", "Valid (according to COSMIC files)?", "COSV identifier", "Mutation IDs", "COSMIC Search URL")))
+		{
+			for (COSMICIdentifierUpdater record : updaters.values().parallelStream().sorted().collect(Collectors.toList()))
+			{
+				// Include a COSMIC Search URL for the identifier in the report, to make it easier for Curators to follow up on identifiers that might need attention.
+				String url;
+				String identifierForUrl = "";
+				if (record.getIdentifier().startsWith("C"))
+				{
+					identifierForUrl = record.getIdentifier();
+				}
+				else
+				{
+					if (record.getSuggestedPrefix() != null)
+					{
+						identifierForUrl = record.getSuggestedPrefix() + record.getIdentifier();
+					}
+					else
+					{
+						identifierForUrl = record.getIdentifier();
+					}
+				}
+				url = "https://cancer.sanger.ac.uk/cosmic/search?q=" + identifierForUrl;
```

Comment:
Could this URL be a global variable? I know it only appears once, but I think cosmic as a resource is growing quite a bit and I wouldn't be surprised if URLs like this change in the future as they continue to update the website, so it would be good for finding it more quickly

Reply:
I'm kinda in favour of this as well.  Maybe a `public static final` constant in the utils class?